internal/driver/driver.go

```diff
@@ -412,8 +415,7 @@ func (d *Driver) Discover() {
 	d.configMu.RUnlock()
 
 	if !discoveryMode.IsValid() {
-		d.lc.Errorf("DiscoveryMode is set to an invalid value: %s. Refusing to do discovery.", discoveryMode)
-		return
+		return fmt.Errorf("discoveryMode is set to an invalid value: %s. Refusing to do discovery", discoveryMode)
```

Comment:
`discoveryMode` is based on `d.config.AppCustom.DiscoveryMode`, which I believe is still uppercase in the configuration, so I would leave it uppercase in the error message: `DiscoveryMode`

Reply:
Updated